File: DataStructures and Algorithms/SelectionSort/selectionsort.py
#Selection sort 

# this function stores the smallest value and stores the index.
def findSmallest(arr):
    smallest = arr[0]
    smallest_index = 0
    for i in range(1, len(arr)):
      if arr[i] < smallest:
          smallest = arr[i]
          smallest_index = i 
    return smallest_index 


 # selectionSort 

def selectionSort(arr):
      newArr = []
      for i in range(len(arr)):
          smallest = findSmallest(arr)
          newArr.append(arr.pop(smallest))
      return newArr 

print(selectionSort([5, 3, 7, 1, 4, 9, 10, 2]))

File: DataStructures and Algorithms/SelectionSort/selectionsort.ts
// Selection sort 


function  selectionSort(arr: number[]): number[]{
    const n = arr.length;

    for (let i = 0; i < n - 1; i++){
        let minIndex = i;

        for (let j = i + 1; j < n; j++){
            if (arr[j] < arr[minIndex]){
                minIndex = j;
            }
        }

        [arr[i], arr[minIndex]] = [arr[minIndex], arr[i]];
    }
    return arr;
}


console.log(selectionSort([4,567,78,9,0,1]))



console.log(34)
